oauthenticator/globus.py

```diff
@@ -109,24 +119,42 @@ def _scope_default(self):
             'urn:globus:auth:scope:transfer.api.globus.org:all'
         ]
 
-    globus_local_endpoint = Unicode(help="""If Jupyterhub is also a Globus endpoint, its endpoint id can
-                be specified here.""").tag(config=True)
+    allow_refresh_tokens = Bool(
+        help="""Allow users to have Refresh Tokens. If Refresh Tokens are not
+        allowed, users must use regular Access Tokens which will expire after
+        a set time. Set to False for increased security, True for increased
+        convenience."""
+    ).tag(config=True)
+
+    def _allow_refresh_tokens_default(self):
+        return True
+
+    globus_local_endpoint = Unicode(help="""If Jupyterhub is also a Globus
+    endpoint, its endpoint id can be specified here.""").tag(config=True)
 
     def _globus_local_endpoint_default(self):
         return os.getenv('GLOBUS_LOCAL_ENDPOINT', '')
 
     def pre_spawn_start(self, user, spawner):
-        """Add tokens to the spawner whenever the spawner starts a notebook. Token names are translated to conform
-        to bash-style variable names. For example, 'transfer.api.globus.org' becomes 'TRANSFER_API_GLOBUS_ORG'"""
-        env_tokens = {
-            self.bashify_variable_name(name): token for name, token in user.auth_state['transfer_tokens'].items()
-        }
+        """Add tokens to the spawner whenever the spawner starts a notebook.
+        This will allow users to create a transfer client:
+        globus-sdk-python.readthedocs.io/en/stable/tutorial/#tutorial-step4
+        """
         if self.globus_local_endpoint:
-            spawner.environment.update({'GLOBUS_LOCAL_ENDPOINT': self.globus_local_endpoint})
-        spawner.environment.update(env_tokens)
-
-    def bashify_variable_name(self, var_name):
-        return var_name.replace('.', '_').replace('-', '_').upper()
+            spawner.environment.update(
+                {'GLOBUS_LOCAL_ENDPOINT': self.globus_local_endpoint}
+            )
+        if user.auth_state.get('globus_data'):
+            globus_data = base64.b64encode(
+                pickle.dumps(user.auth_state['globus_data'])
+            )
+            spawner.environment['GLOBUS_DATA'] = globus_data
```

Comment:
@briedel I think one point that's helpful is that we'll provide a shim layer (a Python module) based on the Globus SDK that handles the tokens in the notebook. This will help users that want to use this out of the box, and also serve as an example to those that want to do more. This is basically what I did during the GlobusWorld demo.